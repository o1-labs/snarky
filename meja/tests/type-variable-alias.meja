let f = fun (x : 'a -> 'b, y : (int as 'a), z : (bool as 'b)) => { };

let g = fun (_: (unit as 'x)) => { };

let h = fun (_: ((bool -> int) -> _ -> _, unit -> unit)) => { };

let x = fun () => {
  let x = (f, g);
  h(x);
};
