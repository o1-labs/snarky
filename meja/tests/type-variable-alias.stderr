File "tests/type-variable-alias.meja", line 9, characters 4-5:
Error: Incompatible types ((bool -> int) -> 'a -> 'b, unit -> unit) and
       ((int -> bool) -> int -> bool -> unit, unit -> unit):
       Cannot unify bool and int
